op-batcher/batcher/driver.go

```diff
@@ -449,9 +463,11 @@ func (l *BatchSubmitter) publishTxToL1(ctx context.Context, queue *txmgr.Queue[t
 		return err
 	}
 
-	if err = l.sendTransaction(ctx, txdata, queue, receiptsCh); err != nil {
-		return fmt.Errorf("BatchSubmitter.sendTransaction failed: %w", err)
-	}
+	daWaitGroup.Add(1)
+	errGroup.Go(func() error {
```

Comment:
1) **In regards to the L1 tip in this function above:**

```go
	l1tip, err := l.l1Tip(ctx)
	if err != nil {
		l.Log.Error("Failed to query L1 tip", "err", err)
		return err
	}

```
This will potentially make a lot of successive `eth_getHeaderByNumber` requests to query for the latest block,  since we spawn the DA send asynchronously and iterate quickly through the outer for loop that calls the `publishTxToL1`.
I don't think it's too important, but if we want to limit this without using a push based notification for new headers we could sleep a little bit after the `errGroup.Go` call. This could also help not spamming the DA node when there is a persistent error returned from `PlasmaDA.SetInput`.

2) **You raised the question in terms of resource caps:**

I think this could easily be added by the `errgroup.SetLimit`. This would make the `.Go` call block when the limit is reached, which is a similar behaviour as before. We could set it to a high, arbitrary number at first and observe the behaviour. A log message for the limit case could be created by using something like:
```go
if spawned := errGroup.TryGo( ... ); !spawned{
  log.Info("limit reached")
  errGroup.Go( ... )
}
```

Reply:
For 1. regarding "This could also help not spamming the DA node when there is a persistent error returned from PlasmaDA.SetInput", I observed this exact behavior in the current op-batcher code, see https://github.com/ethereum-optimism/optimism/pull/11534

For 2. agree, we will need to set a limit since disperser has a rate limiter per sender anyways.

Reply:
I would like to deviate from the limiting offered by the errgroup in one way. Instead of blocking when creating a new goroutine, I would like to skip creating new writes on a higher level (maybe not call `publishStateToL1` when the limit is reached), so that the op-batcher main loop keeps running even in that case.

I assume that blocking the main loop for a longer time will invite all kinds of different problems. So unless we can assure that the blocking is short, we should always return to the main loop quickly.

If we can expect somewhat quick confirmations for all DAs soon (30s confirmations for EigenDA [were mentioned](https://discord.com/channels/1244729134312198194/1260612364865245224/1275919748886171740)), this might be overengineering, though.